kbase-extension/static/kbase/js/common/jobManager.js

```diff
@@ -62,16 +62,24 @@ define([
 
         _isValidMessage(msgType, message) {
             switch (msgType) {
+                case jcm.MESSAGE_TYPE.RUN_STATUS:
+                    return Jobs.isValidRunStatusObject(message);
                 case jcm.MESSAGE_TYPE.STATUS:
-                    return Jobs.isValidBackendJobStateObject(message);
+                    return Object.values(message).every((val) => {
+                        return Jobs.isValidBackendJobStateObject(val);
+                    });
```

Comment:
indexing job messages by job ID means these validators all need to be updated to check each of the values in the message object.